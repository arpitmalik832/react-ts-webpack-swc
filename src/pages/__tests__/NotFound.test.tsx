import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import NotFound from '../NotFound';

describe('NotFound component tests', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders without crashing', () => {
    const { container } = render(<NotFound />);
    expect(container).toBeTruthy();
  });

  it('renders with correct text content', () => {
    const { getByText } = render(<NotFound />);
    expect(getByText('404')).toBeInTheDocument();
  });

  it('renders a div element', () => {
    const { container } = render(<NotFound />);
    const divElement = container.querySelector('div');
    expect(divElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<NotFound />);
    expect(container).toMatchSnapshot();
  });

  it('has correct structure', () => {
    const { container } = render(<NotFound />);
    const divElement = container.firstChild as HTMLElement;

    // Check if it's a div
    expect(divElement.tagName).toBe('DIV');

    // Check text content
    expect(divElement.textContent).toBe('404');

    // Check that it has no child elements (only text)
    expect(divElement.children.length).toBe(0);
  });

  it('is accessible', () => {
    const { container } = render(<NotFound />);
    // Check if the 404 message is visible to screen readers
    expect(container.firstChild).toBeVisible();
  });

  it('has correct role for accessibility', () => {
    const { container } = render(<NotFound />);
    // The main content should be visible and properly structured
    expect(container.firstChild).toBeVisible();
    expect(container.firstChild).toHaveTextContent('404');
  });
});
