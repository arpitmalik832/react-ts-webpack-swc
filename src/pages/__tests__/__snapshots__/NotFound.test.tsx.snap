// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component tests matches snapshot 1`] = `
<div>
  <div>
    404
  </div>
</div>
`;
